src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -57,7 +58,8 @@ public bool CanAddTo(SyntaxNode destination, Solution solution, CancellationToke
             return CanAddTo(destination, solution, cancellationToken, out availableIndices);
         }
 
-        private bool CanAddTo(SyntaxNode destination, Solution solution, CancellationToken cancellationToken, out IList<bool> availableIndices)
+        private bool CanAddTo(SyntaxNode destination, Solution solution, CancellationToken cancellationToken,
+            out IList<bool> availableIndices, Func<Document, bool> isGeneratedCode = null)
         {
```

Comment:
I understand that we use the word "code" to mean file a lot, however in this context we are also generating code so overloading terms by asking "IsGeneratedCode" may get confusing.  Could we say something like "IsGeneratedDocument"?